syntax = "proto3";

package chunkservice;

service ChunkService {
    rpc UploadChunk(UploadChunkRequest) returns (UploadChunkResponse) {}
    rpc DownloadChunk(DownloadChunkRequest) returns (DownloadChunkResponse) {}
}

message UploadChunkRequest {
  string chunk_id = 1; 
  bytes data = 2;      
}

message UploadChunkResponse {
  bool success = 1;
  string message = 2;
}

message DownloadChunkRequest {
  string chunk_id = 1; 
}

message DownloadChunkResponse {
  bytes data = 1; 
}